File: 3225-length-of-longest-subarray-with-at-most-k-frequency/length-of-longest-subarray-with-at-most-k-frequency.py
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        MAP = defaultdict(int)
        mx = 1
        n = len(nums)
        nums = [0] + nums
        l,r = 1,1
        while r<=n:
            MAP[nums[r]]+=1
            while MAP[nums[r]]>k:
                MAP[nums[l]]-=1
                l+=1
            mx = max(mx,r-l+1)
            r+=1
        return mx

File: 3225-length-of-longest-subarray-with-at-most-k-frequency/length-of-longest-subarray-with-at-most-k-frequency.cpp
class Solution {
public:
    int maxSubarrayLength(vector<int>& nums, int k) {
        unordered_map<int,int> MAP;
        int mx = 1;
        int n = nums.size();
        nums.insert(nums.begin(),0);
        int l = 1,r = 1;
        while (r<=n){
            MAP[nums[r]]++;
            while (MAP[nums[r]]>k){
                MAP[nums[l++]]--;
            }
            mx = max(mx,r-l+1);
            r++;
        }
        return mx;
    }
};